src/payment/strategies/opy/opy-payment-strategy.ts

```diff
@@ -0,0 +1,99 @@
+
+    initialize(_options?: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        const { isStoreCreditApplied: useStoreCredit } = this._store.getState().checkout.getCheckoutOrThrow();
+
+        await this._store.dispatch(this._storeCreditActionCreator.applyStoreCredit(useStoreCredit));
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+
+        const { methodId } = payment;
+        const {
+            paymentMethods: { getPaymentMethodOrThrow },
+        } = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId, options));
+        const {
+            order_id: nonce,
+            next_action: { type, formPost: { formPostUrl, formFields } },
+        }: OpyOrderResponse = JSON.parse(String(getPaymentMethodOrThrow(methodId).clientToken));
+
+        if (!formPostUrl || !formFields) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        if (!nonce) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentToken);
+        }
+
+        await this._storefrontPaymentRequestSender.saveExternalId(methodId, nonce);
+
+        try {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({ methodId, paymentData: { nonce } }));
+        } catch (error) {
+            if (error instanceof RequestError && error.body.status === 'additional_action_required') {
+                if (type === 'FormPost') {
+                    const url = new URL(formPostUrl.replace(/\/$/, ''));
+
+                    formFields.forEach(({ fieldName, fieldValue }) => {
+                        url.searchParams.append(fieldName, fieldValue);
+                    });
+
+                    return new Promise(() => window.location.replace(decodeURI(url.href)));
+                }
+
+                throw new NotImplementedError(`Unknown action type: ${type}`);
+            }
+
+            throw error;
+        }
+    }
+
+    finalize(_options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(_options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+}
+interface OpyOrderResponse {
+    order_id?: string;
+    next_action: {
+        type: 'FormPost' | 'WaitForCustomer';
+        formPost: {
+            formPostUrl?: string;
+            formFields?: [{
+                fieldName: string;
+                fieldValue: string;
+            }];
+        };
+    };
+}
```

Comment:
i think you need to add a empty line at the end of each file

Reply:
Thank you! already has one. I think GitHub only warns you when there's no line ending.